runtime/storage_test.go

```diff
@@ -2326,29 +2326,32 @@ func TestRuntimeReferenceOwnerAccess(t *testing.T) {
               prepare(accountA: AuthAccount, accountB: AuthAccount) {
 
                   let testResource <- TestContract.makeTestResource()
-                  let ref = &testResource as &TestContract.TestResource
+                  let ref1 = &testResource as &TestContract.TestResource
 
                   // At this point the resource is not in storage
-                  log(ref.owner?.address)
+                  log(ref1.owner?.address)
 
                   accountA.save(<-testResource, to: /storage/test)
 
                   // At this point the resource is in storage A
-                  log(ref.owner?.address)
+                  accountA.link<&TestContract.TestResource>(/public/test, target: /storage/test)
+                  let ref2 = accountA.getCapability<&TestContract.TestResource>(/public/test).borrow()!
+                  log(ref2.owner?.address)
 
                   let testResource2 <- accountA.load<@TestContract.TestResource>(from: /storage/test)!
 
-                  let ref2 = &testResource2 as &TestContract.TestResource
+                  let ref3 = &testResource2 as &TestContract.TestResource
 
                    // At this point the resource is not in storage
-                  log(ref.owner?.address)
-                  log(ref2.owner?.address)
+                  log(ref3.owner?.address)
 
                   accountB.save(<-testResource2, to: /storage/test)
 
+                  accountB.link<&TestContract.TestResource>(/public/test, target: /storage/test)
+                  let ref4 = accountB.getCapability<&TestContract.TestResource>(/public/test).borrow()!
+
                   // At this point the resource is in storage B
-                  log(ref.owner?.address)
-                  log(ref2.owner?.address)
```

Comment:
Given that `ref2` is a storage reference, it should be possible to keep using it here

Reply:
it would give a dereference error, since the value is moved out of the storage